<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SocketServer</class>
 <widget class="QWidget" name="SocketServer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>922</width>
    <height>585</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_10">
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="spacing">
          <number>5</number>
         </property>
         <property name="leftMargin">
          <number>5</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label_4">
           <property name="minimumSize">
            <size>
             <width>60</width>
             <height>25</height>
            </size>
           </property>
           <property name="text">
            <string>Server A</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_ServerA">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>25</height>
            </size>
           </property>
           <property name="editable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_5">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>25</height>
            </size>
           </property>
           <property name="text">
            <string>Port</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_PortA">
           <property name="minimumSize">
            <size>
             <width>60</width>
             <height>25</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>60</width>
             <height>25</height>
            </size>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>65535</number>
           </property>
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_6">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>25</height>
            </size>
           </property>
           <property name="text">
            <string>protocol</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_ProtocolA">
           <property name="minimumSize">
            <size>
             <width>50</width>
             <height>25</height>
            </size>
           </property>
           <property name="editable">
            <bool>true</bool>
           </property>
           <item>
            <property name="text">
             <string>TCP</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>UDP</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_StartA">
           <property name="minimumSize">
            <size>
             <width>60</width>
             <height>25</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>60</width>
             <height>25</height>
            </size>
           </property>
           <property name="text">
            <string>Start</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <property name="spacing">
          <number>5</number>
         </property>
         <property name="leftMargin">
          <number>5</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label_7">
           <property name="minimumSize">
            <size>
             <width>60</width>
             <height>25</height>
            </size>
           </property>
           <property name="text">
            <string>Server B</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_ServerB">
           <property name="minimumSize">
            <size>
             <width>150</width>
             <height>25</height>
            </size>
           </property>
           <property name="editable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_9">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>25</height>
            </size>
           </property>
           <property name="text">
            <string>Port</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox_PortB">
           <property name="minimumSize">
            <size>
             <width>60</width>
             <height>25</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>60</width>
             <height>25</height>
            </size>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>65535</number>
           </property>
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_8">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>25</height>
            </size>
           </property>
           <property name="text">
            <string>protocol</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_ProtocolB">
           <property name="minimumSize">
            <size>
             <width>50</width>
             <height>25</height>
            </size>
           </property>
           <property name="editable">
            <bool>true</bool>
           </property>
           <item>
            <property name="text">
             <string>TCP</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>UDP</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_StartB">
           <property name="minimumSize">
            <size>
             <width>60</width>
             <height>25</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>60</width>
             <height>25</height>
            </size>
           </property>
           <property name="text">
            <string>Start</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="spacing">
      <number>5</number>
     </property>
     <property name="leftMargin">
      <number>5</number>
     </property>
     <item>
      <widget class="QLabel" name="label_2">
       <property name="minimumSize">
        <size>
         <width>60</width>
         <height>0</height>
        </size>
       </property>
       <property name="text">
        <string>Content</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineEdit">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>25</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>25</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Type</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="comboBox_DataType">
       <property name="minimumSize">
        <size>
         <width>120</width>
         <height>25</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>120</width>
         <height>25</height>
        </size>
       </property>
       <property name="editable">
        <bool>true</bool>
       </property>
       <item>
        <property name="text">
         <string>int</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>hex</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>char</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>float</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>double</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Unsigned int</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>unsigned long int</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>unsigned long long int</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_2">
       <property name="minimumSize">
        <size>
         <width>60</width>
         <height>25</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>60</width>
         <height>25</height>
        </size>
       </property>
       <property name="text">
        <string>+</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QTreeWidget" name="treeWidget_Data">
     <column>
      <property name="text">
       <string>Content</string>
      </property>
     </column>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_9">
     <item>
      <widget class="QLabel" name="label_10">
       <property name="text">
        <string>Recv</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_RecvDataCount">
       <property name="minimumSize">
        <size>
         <width>100</width>
         <height>25</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>100</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_11">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>25</height>
        </size>
       </property>
       <property name="text">
        <string>Send</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_SendDataCount">
       <property name="minimumSize">
        <size>
         <width>100</width>
         <height>25</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>100</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_4">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QCheckBox" name="checkBox_WriteLog">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>25</height>
        </size>
       </property>
       <property name="text">
        <string>Write log</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="lineWidth">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="handleWidth">
      <number>0</number>
     </property>
     <widget class="QTreeWidget" name="treeWidget_Log">
      <attribute name="headerVisible">
       <bool>false</bool>
      </attribute>
      <column>
       <property name="text">
        <string notr="true">1</string>
       </property>
      </column>
     </widget>
     <widget class="QTreeWidget" name="treeWidget_2">
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <column>
       <property name="text">
        <string>Number</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Type</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Content</string>
       </property>
      </column>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
